kore/src/Kore/Step/Simplification/Pattern.hs

```diff
@@ -124,6 +124,7 @@ makeEvaluate sideCondition pattern' =
                 Conditional.splitTerm withSimplifiedCondition
             term' = substitute (toMap $ substitution simplifiedCondition) term
             simplifiedCondition' =
+                -- TODO: is this comment accurate anymore?
```

Comment:
I removed the conversion and the comment because I realized they're outdated: since the `assumedTrue` is `MultiAnd Predicate`, we don't have any distinction between substitutions and other predicates.